### PR TITLE
boop: tests and fixes

### PR DESCRIPTION
**Pull request by antonbobkov**

### CODE REVIEW

File: test/web3_test.erl [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 web3_test() ->
   {ok, Pid} = gen_server:start(web3, [], []),
-  ?debugFmt("started web3 gen_server ~p~n",[Pid]),
+  %% ?debugFmt("started web3 gen_server ~p~n",[Pid]),
```
Comment (antonbobkov) [inline test/web3_test.erl]: (don't look here)

File: test/pending_transactions_test.erl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+get_final_nonces(List) ->
+    {PT1, _} = apply_transactions(List),
+    get_nonce_list(PT1).
+
+get_final_nonces_counters(List) ->
+    {PT1, _} = apply_transactions(List),
+    get_nonce_counter_list(PT1).
+
+no_repeats(List) ->
+    array:to_list(
```
Comment (pdlla) [inline test/pending_transactions_test.erl]: 😱

Comment (pdlla) [reply test/pending_transactions_test.erl]: you could use http://erlang.org/doc/man/lists.html#usort-2

File: test/pending_transactions_test.erl [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	fun(I, N) -> #{nonce => N, player_id => default_id, game_data => I} end, 
+	array:from_list(List)
+       )
+     ).
+    
+
+add_transaction_test() ->
+    ?assertMatch({_, added_new}, apply_transactions([0])),
+
+    ?assertMatch({_, ignored_duplicate}, apply_transactions([0, 0])),
```
Comment (pdlla) [inline test/pending_transactions_test.erl]: how does [0,0] get converted to a TxList?

File: test/supersock_test.erl [hunk 4]
```diff
@@ -6,8 +6,8 @@
 connect() ->
   spawn_link(fun() ->
     {ok, Sock} = gen_tcp:connect("localhost", 8384, [binary, {active, false}]),
-    {ok, Msg} = gen_tcp:recv(Sock, 0),
-    ?debugFmt("recv ~p~n",[Msg]),
+    {ok, _Msg} = gen_tcp:recv(Sock, 0),
```
Comment (pdlla) [inline test/supersock_test.erl]: :o

File: test/pending_transactions_test.erl (hunk 2 above)
Comment (antonbobkov) [reply test/pending_transactions_test.erl]: im not sorting :scream:

File: test/pending_transactions_test.erl (hunk 3 above)
Comment (antonbobkov) [reply test/pending_transactions_test.erl]:
```
fold_transactions(Nonce, Acc) ->
    fold_transactions(#{nonce => Nonce, player_id => default_id}, Acc).
```

File: test/pending_transactions_test.erl (hunk 2 above)
Comment (antonbobkov) [reply test/pending_transactions_test.erl]: `game_data => I` makes all transactions different even if they have same nonce

File: test/pending_transactions_test.erl (hunk 3 above)
Comment (antonbobkov) [reply test/pending_transactions_test.erl]:
```
fold_transactions(T, {PendingTx, _Status}) when is_map(T) ->
    pending_transactions:add_transaction(T, PendingTx);
```

File: test/pending_transactions_test.erl (hunk 2 above)
Comment (pdlla) [reply test/pending_transactions_test.erl]: oh so it takes repeats and makes them not repeats 😱 ?

**Comment by pdlla** [comment]
I started watching hilda on netflix. So cute :o.

**Pull request merged**